src/os/README.md

```diff
@@ -104,3 +108,125 @@ inspiration for several decisions in this library.
 - that said, if an operation is extremely efficient and almost always wanted,
   consider doing it natively, automatically. For example, ints in HSL
   `sockaddr` are always in host byte order, not network byte order.
+
+## Appendix:  C / Hack type mapping cheat sheet
+
+| C type | Hack type |
+|---|---|
+| `int` or `short` as a bit set | `keyset<Flag>` where `Flag` is an `enum` |
+| long-lived `int socket` or `int filedes` | `HH\Lib\OS\FileDescriptor` or other wrapper classes |
+| short-lived `void *` or `struct *` | `shape` or `vec`, see [Appendix: opaque C pointer encoding](#appendix-opaque-c-pointer-encoding) |
```

Comment:
struct *maybe*; `void*`  - when used as a handle - needs a per-system class that stores and frees the void*

Reply:
Agreed, I think the key is its life time, as short-lived structs usually do not hold expensive system resource, and can be created on demand from Hack `shape` or `vec`. I think the on-demand creation approach should apply to any C struct, not only opaque ones.